readthedocs_ext/readthedocs.py

```diff
@@ -126,6 +149,45 @@ def rtd_render(self, template, render_context):
                                                         app.builder.templates)
 
 
+def generate_json_artifacts(app, pagename, templatename, context, doctree):
+    """
+    Generate JSON artifacts for each page.
+
+    This way we can skip generating this in other build step.
+    """
+    try:
+        if not app.config.rtd_generate_json_artifacts:
+            return
```

Comment:
I didn't find a way of using this configuration in the setup step, so I'm checking it here